ghost/core/test/e2e-webhooks/pages.test.js

```diff
@@ -121,6 +121,59 @@ describe('page.* events', function () {
             });
     });
 
+    it('page.deleted event is triggered', async function () {
+        const webhookURL = 'https://test-webhook-receiver.com/page-deleted/';
+        await webhookMockReceiver.mock(webhookURL);
+        await fixtureManager.insertWebhook({
+            event: 'page.deleted',
+            url: webhookURL
+        });
+
+        const res = await adminAPIAgent
+            .post('pages/')
+            .body({
+                pages: [
+                    {
+                        title: 'testing page.deleted webhook',
+                        status: 'published',
+                        published_at: moment().subtract(6, 'hours').toISOString()
+                    }
+                ]
+            })
+            .expectStatus(201);
+
+        const id = res.body.pages[0].id;
+        const pageToDelete = res.body.pages[0];
+
+        await adminAPIAgent
+            .delete('pages/' + id)
+            .expectStatus(204)
+            .expectEmptyBody();
+
+        await webhookMockReceiver.receivedRequest();
+
+        webhookMockReceiver
+            .matchHeaderSnapshot({
+                'content-version': anyContentVersion,
+                'content-length': anyNumber,
+                'user-agent': anyGhostAgent
+            })
+            .matchBodySnapshot({
+                page: {
+                    current: {},
+                    previous: {
+                        id: anyObjectId,
```

Comment:
Did using `buildPreviousPageSnapshotWithTiers` object not work here? Or maybe could have been extended in some way so we could reuse it?

Reply:
With the `buildPreviousPageSnapshotWithTiers` object, I got the following error report:
```
      page.deleted event is triggered:
     "response.body" does not match snapshot.

Snapshot name: `page.* events page.deleted event is triggered 2: [body] 1`

- Expected properties  - 13
+ Received value       +  1

  Object {
    "page": Object {
      "current": Object {},
      "previous": Object {
-       "tiers": Array [
-         Object {
-           "created_at": StringMatching /\\d\{4\}-\\d\{2\}-\\d\{2\}T\\d\{2\}:\\d\{2\}:\\d\{2\}\\\.000Z/,
-           "id": StringMatching /\[a-f0-9\]\{24\}/,
-           "updated_at": StringMatching /\\d\{4\}-\\d\{2\}-\\d\{2\}T\\d\{2\}:\\d\{2\}:\\d\{2\}\\\.000Z/,
-         },
-         Object {
-           "created_at": StringMatching /\\d\{4\}-\\d\{2\}-\\d\{2\}T\\d\{2\}:\\d\{2\}:\\d\{2\}\\\.000Z/,
-           "id": StringMatching /\[a-f0-9\]\{24\}/,
-           "updated_at": StringMatching /\\d\{4\}-\\d\{2\}-\\d\{2\}T\\d\{2\}:\\d\{2\}:\\d\{2\}\\\.000Z/,
-         },
-       ],
-       "updated_at": StringMatching /\\d\{4\}-\\d\{2\}-\\d\{2\}T\\d\{2\}:\\d\{2\}:\\d\{2\}\\\.000Z/,
+       "updated_at": "2022-10-31T09:51:08.000Z",
      },
    },
  }
```

I can try to extend it.

Reply:
I extended `buildPreviousPageSnapshotWithTiers` in the next commit, but I'm not sure if it fully matches the original intent. One thing at least is that the response for the pages.deleted webhook does not return any tiers it seems, so it would be consistent in that respect.

Reply:
Oh I see. Seems like we need a little bit more thinking in how we name these "matcher building" methods. I think the approach is good enough for now 👍